spec/app/domain/authentication/authn-oidc/discover_oidc_provider_spec.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+RSpec.describe Authentication::AuthnOidc::AuthenticateIdToken::DiscoverOIDCProvider do
+
+  let (:test_provider_uri) { "test-provider-uri" }
+  let (:test_error) { "test-error" }
+  let (:mock_provider) { "test-provider" }
+
+  def mock_discovery_provider(error:)
+    double('discovery_provider').tap do |discovery_provider|
+      if error
+        allow(discovery_provider).to receive(:discover!)
+                                       .and_raise(error)
+      else
+        allow(discovery_provider).to receive(:discover!)
+                                       .and_return(mock_provider)
+      end
+    end
+  end
```

Comment:
The following comment is more for spreading knowledge than a change you need to make, which in this particular case I don't think is very important, even though "by the book" best practice would recommend it.

[Don't mock what you don't own](https://github.com/testdouble/contributing-tests/wiki/Don%27t-mock-what-you-don%27t-own)

In the above, we're mocking `OpenIDConnect::Discovery::Provider::Config.discover!` ([link](https://www.rubydoc.info/gems/openid_connect/0.8.2/OpenIDConnect/Discovery/Provider/Config)), which we don't own.

The best practice solution would be to wrap that class with our own, and mock _that_.  This would ensure that the details of this class and of its return values (`jwks` method) wouldn't leak into our code.

Why don't I think this matters too much in this particular case?

As I mentioned in another comment, `DiscoverOidcProvider` and `FetchOidcSecrets` are effectively providing the encapsulation we care about, and I think it's unlikely we'll be using `OpenIDConnect::Discovery::Provider::Config` anywhere *besides* `DiscoverOidcProvider`.